### PR TITLE
More layering

### PR DESCRIPTION
**Pull request by anthonydresser**

### CODE REVIEW

File: src/sql/platform/capabilities/common/capabilitiesServiceImpl.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Emitter } from 'vs/base/common/event';
```
Comment (anthonydresser) [inline src/sql/platform/capabilities/common/capabilitiesServiceImpl.ts]: This is the biggest change. I rewrote the capabilities to not have a cache and not using a registry. It cleans up the dependency chain a good bit and makes the code cleaner. I didn't see any side effects in my testing. It should work exactly the same.

File: src/sql/workbench/services/query/common/dataService.ts [hunk 2]
```diff
@@ -3,32 +3,38 @@
...
  * query running and grid interaction communication for a single URI.
  */
 export class DataService {
 
-	public queryEventObserver: Subject<any>;
-	public gridContentObserver: Subject<any>;
+	public fireQueryEvent(event: any) {
+		this._queryEvents.fire(event);
+	}
+	private readonly _queryEvents = new Emitter<any>();
```
Comment (anthonydresser) [inline src/sql/workbench/services/query/common/dataService.ts]: Rewrote these to use events over rxjs

File: src/sql/workbench/services/query/common/queryModel.ts [hunk 3]
```diff
@@ -52,9 +51,9 @@ export interface IQueryModelService {
 	getQueryRunner(uri: string): QueryRunner | undefined;
 
 	getQueryRows(uri: string, rowStart: number, numberOfRows: number, batchId: number, resultId: number): Promise<ResultSetSubset | undefined>;
-	runQuery(uri: string, selection: ISelectionData | undefined, queryInput: QueryEditorInput, runOptions?: ExecutionPlanOptions): void;
-	runQueryStatement(uri: string, selection: ISelectionData | undefined, queryInput: QueryEditorInput): void;
-	runQueryString(uri: string, selection: string | undefined, queryInput: QueryEditorInput): void;
+	runQuery(uri: string, selection: ISelectionData | undefined, runOptions?: ExecutionPlanOptions): void;
```
Comment (anthonydresser) [inline src/sql/workbench/services/query/common/queryModel.ts]: removed unused parameter

File: src/sql/workbench/contrib/connection/common/connectionProviderExtension.ts [hunk 4]
```diff
@@ -166,24 +131,42 @@ const ConnectionProviderContrib: IJSONSchema = {
 	required: ['providerId']
 };
 
-ExtensionsRegistry.registerExtensionPoint<ConnectionProviderProperties | ConnectionProviderProperties[]>({ extensionPoint: 'connectionProvider', jsonSchema: ConnectionProviderContrib }).setHandler(extensions => {
+const connectionProviderExtPoint = ExtensionsRegistry.registerExtensionPoint<ConnectionProviderProperties | ConnectionProviderProperties[]>({ extensionPoint: 'connectionProvider', jsonSchema: ConnectionProviderContrib });
 
-	function handleCommand(contrib: ConnectionProviderProperties, extension: IExtensionPointUser<any>) {
-		connectionRegistry.registerConnectionProvider(contrib.providerId, contrib);
-	}
+class ResourceLabelFormattersHandler implements IWorkbenchContribution {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/connection/common/connectionProviderExtension.ts]: class name doesn't seem correct

**Pull request merged**